### ISSUE TITLE
Cannot deal with undefined variables

### ISSUE THREAD

**Issue by Icenowy** [oilshell/oil#4]
```
icenowy@x220i [ oil@master ] $ bin/osh
osh$ $a
Traceback (most recent call last):
  File "bin/osh", line 374, in <module>
    sys.exit(main(sys.argv))
  File "bin/osh", line 364, in main
    return OshMain(main_argv)
  File "bin/osh", line 274, in OshMain
    InteractiveLoop(opts, ex, c_parser, w_parser, line_reader)
  File "bin/osh", line 100, in InteractiveLoop
    status, cflow = ex.ExecuteTop(node)
  File "/home/icenowy/git-repos/oil/bin/../core/cmd_exec.py", line 635, in ExecuteTop
    status, cflow = self.Execute(node)
  File "/home/icenowy/git-repos/oil/bin/../core/cmd_exec.py", line 663, in Execute
    raise AssertionError('Error evaluating words: %s' % err)
AssertionError: Error evaluating words: ['Undefined variable a', 'Error evaluating word part (SimpleVarSub token:(token id:VSub_Name val:"$a" span_id:0))', 'Error evaluating word (CompoundWord parts:[(SimpleVarSub token:(token id:VSub_Name val:"$a" span_id:0))])']
```

But on a standard POSIX sh, the value of undefined variables are just nothing.

**Comment by andychu** [comment]
Yes, this is a bug I just introduced in the process of making the AST more detailed.  To convert osh to oil, I need to represent the different between $a and ${a}, which doesn't matter from an execution point of view.  So the executor doesn't deal properly with $a (I think ${a}) should work.

After the AST is done evolving (based on three use cases I outlined at the end of this post:  http://www.oilshell.org/blog/2017/01/06.html ) I need to make another pass on the executor to fix up stuff like this.

Thanks for trying it!

**Comment by andychu** [comment]
I fixed this with commit cb22624ad0d16544e8be2883e9ba870b47e47070.  That fixed the way we processed $a and ${a}  -- i.e. the difference introduced by the Lossless Syntax Tree.

Now it complains about not having one argument, which isn't right either, but we're making progress :)

If you are interested in helping let me know, or subscribe to oil-dev: http://lists.oilshell.org/private.cgi/oil-dev-oilshell.org/2017-March/thread.html



```
$ bin/osh
osh$ $a
Traceback (most recent call last):
  File "bin/osh", line 378, in <module>
    sys.exit(main(sys.argv))
  File "bin/osh", line 368, in main
    return OshMain(main_argv)
  File "bin/osh", line 278, in OshMain
    InteractiveLoop(opts, ex, c_parser, w_parser, line_reader)
  File "bin/osh", line 100, in InteractiveLoop
    status, cflow = ex.ExecuteTop(node)
  File "/home/andy/git/oil/bin/../core/cmd_exec.py", line 866, in ExecuteTop
    status, cflow = self.Execute(node)
  File "/home/andy/git/oil/bin/../core/cmd_exec.py", line 852, in Execute
    status, cflow = self._Execute(node)
  File "/home/andy/git/oil/bin/../core/cmd_exec.py", line 661, in _Execute
    thunk = self._GetThunkForSimpleCommand(argv, more_env)
  File "/home/andy/git/oil/bin/../core/cmd_exec.py", line 503, in _GetThunkForSimpleCommand
    assert argv, "Need at least one arugment"
AssertionError: Need at least one arugment
```
